src/modules/csm/csm.py

```diff
@@ -273,11 +273,14 @@ def current_frame_range(self, blockstamp: BlockStamp) -> tuple[EpochNumber, Epoc
         if converter.frame_config.initial_epoch == far_future_initial_epoch:
             raise ValueError("CSM oracle initial epoch is not set yet")
 
-        l_ref_slot = self.w3.csm.get_csm_last_processing_ref_slot(blockstamp)
+        l_ref_slot = last_processing_ref_slot = self.w3.csm.get_csm_last_processing_ref_slot(blockstamp)
         r_ref_slot = initial_ref_slot = self.get_initial_ref_slot(blockstamp)
 
+        if last_processing_ref_slot > blockstamp.slot_number:
+            raise ValueError(f"{last_processing_ref_slot=} > {blockstamp.slot_number=}")
```

Comment:
let's add a bit details here. like " X more than Y. probably, expected slot isn't reached yet"

Reply:
Changed to execution layer provider error instead. `last_processing_ref_slot` can't be greater than current slot, and it's pure requests to the Oracle contract, so it's rather something wrong with EL.